www/advanced-http.js

```diff
@@ -114,12 +114,20 @@ var publicInterface = {
   head: function (url, params, headers, success, failure) {
     return publicInterface.sendRequest(url, { method: 'head', params: params, headers: headers }, success, failure);
   },
-  uploadFile: function (url, params, headers, filePath, name, success, failure) {
-    return publicInterface.sendRequest(url, { method: 'upload', params: params, headers: headers, filePath: filePath, name: name }, success, failure);
+  uploadFiles: function (url, data, headers, filePaths, name, success, failure) {
+    return publicInterface.sendRequest(url, { method: 'upload', data: data, headers: headers, filePaths: filePaths, name: name }, success, failure);
   },
   downloadFile: function (url, params, headers, filePath, success, failure) {
     return publicInterface.sendRequest(url, { method: 'download', params: params, headers: headers, filePath: filePath }, success, failure);
   }
 };
 
 module.exports = publicInterface;
+
+if (!cordova.plugins) {
```

Comment:
I'm not sure about those lines [127-133]... Shouldn't it be that cordova itself registers plugin in `cordova.plugins` automatically? It looks it was working like this till now, or?